src/Core/Settings.h

```diff
@@ -680,6 +680,7 @@ class IColumn;
     M(Bool, query_plan_aggregation_in_order, true, "Use query plan for aggregation-in-order optimisation", 0) \
     M(Bool, query_plan_remove_redundant_sorting, true, "Remove redundant sorting in query plan. For example, sorting steps related to ORDER BY clauses in subqueries", 0) \
     M(Bool, query_plan_remove_redundant_distinct, true, "Remove redundant Distinct step in query plan", 0) \
+    M(Bool, query_plan_optimize_lazy_projection, false, "Use query plan for lazy projection optimisation", 0) \
```

Comment:
Better to use `query_plan_optimize_lazy_materialization`

Reply:
Thank you again for taking the time to review my work. I have renamed it to query_plan_optimize_lazy_materialization.